cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -106,7 +111,7 @@ public void processMessage(MetadataMessage message, StructuredTableContext conte
 
     LOG.trace("Processing message: {}", message);
     EntityId entityId = message.getEntityId();
-
+    long count = 0;
```

Comment:
This seems not needed. You can just have `emitApplicationCountMetrics()` method to get the count from the `appMetadataStore` and emit the metrics.

Reply:
fixed